optaplanner-core/src/main/java/org/optaplanner/core/impl/score/stream/bavet/bi/BavetJoinBridgeBiTuple.java

```diff
@@ -16,19 +16,14 @@
 
 package org.optaplanner.core.impl.score.stream.bavet.bi;
 
-import java.util.LinkedHashSet;
-import java.util.Set;
-
 import org.optaplanner.core.impl.score.stream.bavet.common.BavetJoinBridgeTuple;
-import org.optaplanner.core.impl.score.stream.bavet.common.BavetJoinTuple;
 
 public final class BavetJoinBridgeBiTuple<A, B> extends BavetAbstractBiTuple<A, B>
         implements BavetJoinBridgeTuple {
 
     protected final BavetAbstractBiTuple<A, B> parentTuple;
     private final BavetJoinBridgeBiNode<A, B> node;
 
-    protected Set<BavetJoinTuple> childTupleSet = new LinkedHashSet<>(); // TODO capacity
```

Comment:
This is intresting. The Set becomes a List (which is a good thing I think.)

Did this bring a performance gain? If not, is it possible it was offset by the unmodifiable wrapping?

Reply:
I'll re-benchmark once this is merged. But I don't really think this should result in such a big gain - it's just inside one particular tuple, not inside all of them.